main.c

```diff
@@ -693,12 +694,103 @@ _install_transaction (Pool         *pool,
     {
       Id p = installedq.elements[x];
       queue_pushunique (pile, p);
-      g_debug ("%*c - %s", indent, ' ', pool_solvid2str (pool, p));
+      const gchar *solvable = pool_solvid2str (pool, p);
+      g_debug ("%*c - %s", indent, ' ', solvable);
+      /* Non-modules are immediately resolved as resolved, since for RPM the
+       * result would not change if done again. However for modules we need to
+       * make sure we look at all combinations. */
+      if (!g_str_has_prefix (solvable, "module:"))
+        map_set (tested, p);
     }
 
   return TRUE;
 }
 
+/**
+ * mask_bare_rpms:
+ * @pool: initialized pool
+ *
+ * For each available modular package, find all bare RPMs with the same name,
+ * and mark them as not considered.
+ */
+static void
+mask_bare_rpms(Pool *pool)
+{
+  /* Get array of all existing modular packages. */
+  Id *modular_packages = pool_whatprovides_ptr (pool, pool_str2id (pool, MODPKG_PROV, 1));
+
+  /* Filter it to keep only available packages. A package that is not
+   * considered should not mask anything. */
+  g_auto(Queue) available_modular_pkgs;
+  queue_init(&available_modular_pkgs);
+  for (Id *pp = modular_packages; *pp; pp++)
+    {
+      if (map_tst (pool->considered, *pp))
+        {
+          queue_push2 (&available_modular_pkgs, SOLVER_SOLVABLE | SOLVER_INSTALL, *pp);
+        }
+    }
+
+  for (int i = 0; i < available_modular_pkgs.count; i++)
+    {
+      Id pp = available_modular_pkgs.elements[i];
+      Solvable *modpkg = pool_id2solvable (pool, pp);
+      const gchar *name = pool_id2str (pool, modpkg->name);
+
+      g_auto(Queue) sel;
+      queue_init (&sel);
+      selection_make (pool, &sel, name, SELECTION_NAME);
+      if (!sel.count)
+        {
+          /* This should never happen, at least one package
+           * should match (the modular one). */
+          continue;
+        }
+      g_auto(Queue) q;
+      queue_init (&q);
+      selection_solvables (pool, &sel, &q);
+
+      for (int j = 0; j < q.count; j++)
+        {
+          Id p = q.elements[j];
+          if (!queue_contains (&available_modular_pkgs, p))
+            {
+              map_clr (pool->considered, p);
+            }
+        }
+    }
+}
+
+/**
+ * disable_module:
+ * @pool: initialized pool
+ * @module: Id of the module to be disabled
+ *
+ * Set the module and all packages in it as not considered. The packages would
```

Comment:
I don't like removal of considering packages. I think we should just disable module solvable. Can you rescribe case which would be impossible to do if packages from module are not disabled?

Reply:
It would be possible to not disable the packages, but it would mean in `mask_bare_rpms` the way of selecting which modular packages are visible would have to change. Basically what's needed there is *list of all modular packages than are in available modules*. Current implementation asks for everything that provides *modular-package()* and then removes packages that are not considered. How can I get the list of packages better?

Reply:
hmmm, there is some function `pool_is_trivially_installable` or smth like that... it might do the trick.

I will look at it when I have time